pageserver/src/tenant/timeline.rs

```diff
@@ -5010,6 +5036,10 @@ impl Timeline {
         // 1. it is older than cutoff LSN;
         // 2. it is older than PITR interval;
         // 3. it doesn't need to be retained for 'retain_lsns';
+
+        // TODO(yuchen): we could consider current retain_lsns as infinite leases.
+        // 3.5. it does not need to be kept for LSNs holding valid leases (logic is very similar to retain_lsns)
```

Comment:
nit: Let's update the numbers on existing lines here rather than inserting `.5` items

Reply:
addressed